types/googlemaps/reference/info-window.d.ts

```diff
@@ -1,78 +1,208 @@
+         */
+        position_changed: [];
+
+        /**
+         * This event is fired when the InfoWindow's zIndex changes.
+         * @see {@link https://developers.google.com/maps/documentation/javascript/reference/info-window#InfoWindow.zindex_changed Maps JavaScript API}
+         * @see {@link InfoWindowOptions#zIndex}
+         * @see {@link InfoWindow#getZIndex}
+         * @see {@link InfoWindow#setZIndex}
+         */
+        zindex_changed: [];
+    }
+
     /**
      * An overlay that looks like a bubble and is often connected to a marker.
      * This class extends MVCObject.
+     * @see {@link https://developers.google.com/maps/documentation/javascript/reference/info-window#InfoWindow Maps JavaScript API}
      */
     class InfoWindow extends MVCObject {
         /**
-         * Creates an info window with the given options. An InfoWindow can be
-         * placed on a map at a particular position or above a marker,
-         * depending on what is specified in the options. Unless auto-pan is
-         * disabled, an InfoWindow will pan the map to make itself visible
-         * when it is opened. After constructing an InfoWindow, you must call
-         * open to display it on the map. The user can click the close button
-         * on the InfoWindow to remove it from the map, or the developer can
-         * call close() for the same effect.
+         * Creates an info window with the given options. An {@link InfoWindow} can be placed on a map at a particular
+         * position or above a marker, depending on what is specified in the options. Unless auto-pan is disabled, an
+         * {@link InfoWindow} will pan the map to make itself visible when it is opened. After constructing an
+         * {@link InfoWindow}, you must call open to display it on the map. The user can click the close button on the
+         * {@link InfoWindow} to remove it from the map, or the developer can call {@link close}() for the same effect.
+         * @see {@link https://developers.google.com/maps/documentation/javascript/reference/info-window#InfoWindow.constructor Maps JavaScript API}
          */
         constructor(opts?: InfoWindowOptions);
-        /** Closes this InfoWindow by removing it from the DOM structure. */
+
+        /**
+         * @see {@link InfoWindowHandlerMap#closeclick closeclick} event
+         * @see {@link InfoWindowHandlerMap#content_changed content_changed} event
+         * @see {@link InfoWindowHandlerMap#domready domready} event
+         * @see {@link InfoWindowHandlerMap#position_changed position_changed} event
+         * @see {@link InfoWindowHandlerMap#zindex_changed zindex_changed} event
+         */
+        addListener<N extends keyof InfoWindowHandlerMap>(
+            eventName: N,
+            handler: MVCEventHandler<this, InfoWindowHandlerMap[N]>,
+        ): MapsEventListener;
+
+        addListener(eventName: string, handler: MVCEventHandler<this, unknown[]>): MapsEventListener;
```

Comment:
What is this general overload good for? For other event types of `MVCObject`? Couldn't the first one also include the keys of `MVCObject` events?

Reply:
In the [`Map`object](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/4b51763cdda43057b20a89a00b61a1f34d6fe4d5/types/googlemaps/reference/map.d.ts#L168-L169), I used this so as not to break backward compatibility. [`MVCObject`](https://developers.google.com/maps/documentation/javascript/reference/event#MVCObject) has no events.

I can replace it with:

```suggestion
        /** @deprecated */
        addListener(eventName: string, handler: MVCEventHandler<this, any[]>): MapsEventListener;
```

Do you propose to remove it altogether?